lib/ansible/modules/cloud/amazon/aws_elasticbeanstalk_app.py

```diff
@@ -0,0 +1,265 @@
+    if app_name is not None:
+        apps = ebs.describe_applications(ApplicationNames=[app_name])
+    else:
+        apps = ebs.describe_applications()
+
+    return apps.get("Applications", [])
+
+
+def check_app(ebs, app, module):
+    app_name = module.params['app_name']
+    description = module.params['description']
+    state = module.params['state']
+    terminate_by_force = module.params['terminate_by_force']
+
+    result = {}
+
+    if state == 'present' and app is None:
+        result = dict(changed=True, output="App would be created")
+    elif state == 'present' and app.get("Description", None) != description:
+        result = dict(changed=True, output="App would be updated", app=app)
+    elif state == 'present' and app.get("Description", None) == description:
+        result = dict(changed=False, output="App is up-to-date", app=app)
+    elif state == 'absent' and app is None:
+        result = dict(changed=False, output="App does not exist")
+    elif state == 'absent' and app is not None:
+        result = dict(changed=True, output="App will be deleted", app=app)
+    elif state == 'absent' and app is not None and terminate_by_force is True:
+        result = dict(changed=True, output="Running environments terminated before the App will be deleted", app=app)
+
+    module.exit_json(**result)
+
+
+def filter_empty(**kwargs):
+    retval = {}
+    for k, v in kwargs.items():
+        if v:
+            retval[k] = v
+    return retval
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+
+    argument_spec.update(
+        dict(
+            app_name=dict(aliases=['name'], type='str', required=False),
+            description=dict(),
+            state=dict(choices=['present', 'absent', 'list'], default='present'),
+            terminate_by_force=dict(type='bool', default=False, required=False)
+        )
+    )
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    app_name = module.params['app_name']
+    description = module.params['description']
+    state = module.params['state']
+    terminate_by_force = module.params['terminate_by_force']
+
+    if app_name is None:
+        if state != 'list':
+            module.fail_json(msg='Module parameter "app_name" is required if "state" is not "list"')
+
+    result = {}
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+
+    if region:
+        ebs = boto3_conn(module, conn_type='client', resource='elasticbeanstalk',
+                         region=region, endpoint=ec2_url, **aws_connect_params)
+    else:
+        module.fail_json(msg='region must be specified')
+
+    app = describe_app(ebs, app_name)
+
+    if module.check_mode and state != 'list':
+        check_app(ebs, app, module)
+        module.fail_json(msg='ASSERTION FAILURE: check_app() should not return control.')
```

Comment:
This is not supporting check mode. Either support check mode properly, or don't claim to support it.